tests/test_modeling_marian.py

```diff
@@ -44,6 +65,23 @@ def model(self):
         else:
             return model
 
+    def _test_repl_generate_batch(self):
+        model_inputs: dict = self.tokenizer.prepare_translation_batch(src_texts=self.src_text).to(torch_device)
+        self.assertEqual(self.model.device, model_inputs["input_ids"].device)
+        generated_ids = self.model.generate(
+            model_inputs["input_ids"],
+            attention_mask=model_inputs["attention_mask"],
+            length_penalty=1.0,  # same as C++
+            num_beams=2,  # 6 is the default
+            # no_repeat_ngram_size=1,
+            # bad_words_ids=[[self.tokenizer.pad_token_id]],
+            decoder_start_token_id=self.tokenizer.pad_token_id,  # mimics 0 embedding at first step.
+        )
+        generated_words = self.tokenizer.decode_batch(generated_ids, skip_special_tokens=True)
```

Comment:
Just noticed that there is a decode_batch function in the Marian tokenizer. I think we should remove this or is it planned that this function is also in `tokenizer_utils.py` ? I think we should keep each models API in general as small as possible and try to only expose the common `_utils` functions. 

IMO, the user will get used ot the function `decode_batch` and wonder why it does not exist for other tokenizers and not for the FastTokenizer either. Or we should implement a general `decode_batch` function in `tokenization_utils` , but not for an individual tokenizer.

Reply:
@thomwolf @LysandreJik

Reply:
Would love to hoist it up to PretrainedTokenizer.
 https://github.com/huggingface/transformers/pull/4159 does that.

Reply:
Fine for me.

Maybe let's use the same naming convention with `batch_decode` indeed (not that I'm a big fan of the one we have right now for `batch_encode` but it's better to be consistent indeed)